hammerblade/torch/kernel/kernel_index_select.cpp

```diff
@@ -0,0 +1,52 @@
+//====================================================================
+// Dot product kernel
+// 03/06/2020 Lin Cheng (lc873@cornell.edu)
+//====================================================================
+
+#include <kernel_common.hpp>
+
+extern "C" {
+
+  __attribute__ ((noinline))  int tensorlib_index_select(
+          hb_tensor_t* self_p,
+          hb_tensor_t* result_p,
+          hb_tensor_t* index_p) {
+
+    HBTensor<float> self(self_p);
+    HBTensor<float> result(result_p);
+    HBTensor<int32_t> index(index_p);
+    float* self_data = (float*)self.data_ptr();
+    float* result_data = (float*)result.data_ptr();
+    int32_t* index_data = (int32_t*)index.data_ptr();
+
+    bsg_cuda_print_stat_kernel_start();
+
+    auto size0 = self.dim(0); // this should really be size(0)
+    auto rowsize = size0 == 0 ? 1 : self.numel() / size0;
```

Comment:
I copied this line from CPU impl, but I don't think it makes much sense ... because if size0 is 0, the tensor has to be tensor([]), and you can't really "select" anything from it. I guess it is here just to prevent divide-by-zero exception.

Reply:
Wouldn't size be 0 for 0-d tensors (scalars)?

Reply:
there is a check before this one which disallows selecting from 0-d tensors.